### DISCUSSION TITLE
Local app different from app deployed to github.io

### DISCUSSION THREAD

**Discussion by sfederici** [community/community#21974] [Other Feature Feedback, Questions, & Ideas]
<p>I forked a repository (my fork is <a href="https://github.com/sfederici/scratch-gui" class="inline-onebox" rel="noopener nofollow ugc">GitHub - sfederici/scratch-gui: Graphical User Interface for creating and running Scratch 3.0 projects.</a>), applied some changes to it, then locally ran the app with</p>
<p>npm start</p>
<p>Then I also deployed the app, from the same folder, to github DOT io with</p>
<p>npm run deploy</p>
<p>The deployed app is visibile at <a href="https://sfederici.github.io/scratch-gui/" class="inline-onebox" rel="noopener nofollow ugc">Scratch 3.0 GUI</a></p>
<p>Both the local and deployed app run well, but the <strong>deployed app</strong> is <strong>from a previous version</strong> of the app. Indeed, when I run it locally, I can see more elements (basically, for the Scratch-savvy, both the “move 100 steps” in the blue “Motion” blocks AND a “forever if” block in the yellow “Control” blocks in the left column of colored blocks; instead, in the deployed app, I can’t see the yellow “forever if” Control block).</p>
<p>What can I have made wrong if the local and the online version are different?</p>

**Comment by sfederici** [comment, marked as answer]
<p>I realized now that the deployed app is the one that has been built. I had just started it.</p>

**Comment by jfullstackdev** [comment]
<p>meaning, it’s now fixed?</p>

**Comment by sfederici** [comment]
<p>Yes, thanks. Building the app before deploying it fixed the problem.</p>

**Comment by jfullstackdev** [comment]
<p>yup that’s quite a common error, thinking that GitHub Page will run everything without first building it</p>

**Comment by queenofcorgis** [comment]
<p><a class="mention" href="/sfederici">@sfederici</a> I marked your answer as the solution so members can easily find the outcome, thanks for letting us know the cause and thank you <a class="mention" href="/jdevstatic">@jdevstatic</a> for jumping in!</p>
